redisvl/providers/openai.py

```diff
@@ -0,0 +1,48 @@
+from typing import Dict, List, Optional
+
+import openai
+
+from redisvl.providers.base import BaseProvider
+
+
+class OpenAIProvider(BaseProvider):
+    def __init__(self, model: str, api_config: Optional[Dict] = None):
+        dims = 1536
+        super().__init__(model, dims, api_config)
+        if not api_config:
+            raise ValueError("OpenAI API key is required in api_config")
+
+        openai.api_key = api_config.get("api_key", None)
+        self._model_client = openai.Embedding
+
+    def embed_many(
```

Comment:
should we look to make this immune to rate limit issues? One pattern I have seen is to use `tenacity` and their exponential backoff decorator.

Reply:
@tylerhutcherson can you point me to or write a code sample for this?